docs/shared/concepts/customizing-inputs.md

```diff
@@ -0,0 +1,128 @@
+# Customizing Inputs
```

Comment:
I think it's good. I'd mention a few things:
1. That by default inputs ["default", "^default"]. So it's somewhat conservative, which means that it will work correclty but will invalidate cache too many times, where it could have been reused.
2. I'd also point out that it's ok for a task to depend on more inputs (like the default case). It's less efficient but works. So you don't have to define your inputs in the most precise way etc.